doc/index.md

```diff
@@ -1032,68 +1050,153 @@ On success, returns the response [*headers*](#http.headers) and a [*stream*](#st
 
 ## http.server
 
-This interface is **unstable**.
+The http.server object is used to listen on sockets and respond to all types of HTTP requests. Each client request is threaded and independent of the other client requests. An `http.server` is instantiated by passing a number of options including an `onstream` response handler that processes the HTTP requests and provides a stream mechanism for response. `onstream` can also be used for testing and upgrading a request, with HTTP 1.1 to WebSockets being the notible example. The server can be made non-blocking when used in conjunction with cqueues. 
+
+### `server.listen(options)` <!-- --> {#http.server.listen}
+
+Create a new instance of an HTTP Server by passing in table `options`:
+
+  - `.client_timeout`: Timeout (in seconds) to wait for client to send first bytes and/or complete TLS handshake (default: 10)
+  - `.cq` (optional): A cqueues controller to use
+  - `.ctx`: An `openssl.ssl.context` object to use for tls connections. If `nil` is passed, a self-signed context will be generated
+  - `.family`: (optional) Protocol family
+  - `.host`: Address to bind to (required if not `.path`)
+  - `.port`: Port to bind to (optional if tls isn't `nil`, in which case defaults to 80 for `.tls == false` or 443 if `.tls == true`)
+  - `.path`: Path to UNIX socket (required if not `.host`)
+  - `.v6only`: Allow ipv6 only (no ipv4-mapped-ipv6)
+  - `.mode`: (optional) `fchmod` or `chmod` socket after creating UNIX domain socket
+  - `.mask`: (optional) Set and restore umask when binding UNIX domain socket
+  - `.unlink`: (optional) Unlink socket path before binding
+  - `.max_concurrent`: Maximum number of connections to allow live at a time (default: infinity)
+  - `.onerror`: Function that will be called when an error occurs (default: do nothing). See [server:onerror()](#http.server:onerror)
+  - `.onstream`: Callback function for handling a new client request. The function requires parameters for [*server*](#http.server) and [*stream*](#stream) and if your function throws an error it will be reported from [*step*](#http.server:step) or [*loop*](#http.server:loop)
+  - `.reuseaddr`: (optional) Turn on SO_REUSEADDR flag
+  - `.reuseport`: (optional) Turn on SO_REUSEPORT flag
+  - `.tls`: `nil`: Allow both tls and non-tls connections
+  -         `true`: Allows tls connections only
+  -         `false`: Allows non-tls connections only
+  - `.version`: The http version to allow to connect (default: any)
+
+
+#### Example {#http.server.listen-example}
 
-### `listen(options)` <!-- --> {#http.server.connect}
+```lua
+
```

Comment:
no newline required after a \`\`\`

Also, the below code is indented too far (it has double tabs)